upstream-sha256: 62fded40a856e0a887a64a34d295c4a7e6da34ffc6067100720ae7e707fa1b2d
actual-sha256: a8e833972b9f58191c330f78587cdca0a7230cb41dfa982c860ffcc10e81a575
--- upstream.expected
+++ tolk.actual
@@ -20,7 +20,7 @@
   float val4 = (*(data4_16+(lidx1+4)));
   float val5 = (*(data4_16+(lidx1+8)));
   float4 val6 = (*((float4*)((data3_16+alu0))));
-  *(data0_16+(lidx1+alu0)) = ((*(buf0+0))+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3));
+  *(data0_16+(lidx1+alu0)) = ((*(buf0+0))+(0.0f+(val6.x*val2)+(val6.y*val4)+(val6.z*val5)+(val6.w*val3)));
 }
 ---
 typedef long unsigned int size_t;
@@ -35,7 +35,7 @@
   __attribute__((shared, aligned(16)))float buf4[16];
   int lidx0 = __ockl_get_local_id(0); /* 16 */
   float val0 = (*(data1_16+lidx0));
-  *(buf4+lidx0) = (val0*val0);
+  *(buf4+lidx0) = (0.0f+(val0*val0));
   __builtin_amdgcn_fence(__ATOMIC_RELEASE, "workgroup");__builtin_amdgcn_s_barrier();__builtin_amdgcn_fence(__ATOMIC_ACQUIRE, "workgroup");
   *(buf0+0) = 0.0f;
   for (int Ridx106 = 0; Ridx106 < 16; Ridx106++) {
